src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/magical/talismans/Talisman.java

```diff
@@ -219,7 +220,10 @@ private static void consumeItem(Inventory inv, Talisman talisman) {
     }
 
     private static Player getPlayerByEventType(Event e) {
-        if (e instanceof EntityDeathEvent) {
+        if (e instanceof PlayerDeathEvent) {
+            return ((PlayerDeathEvent) e).getEntity().getPlayer();
```

Comment:
`getPlayer()` is redundant here `getEntity()` already returns a Player and `getPlayer()` literally just does `return this;` for players.